extension/src/androidMain/kotlin/com/dimension/maskbook/extension/utils/MessageChannel.kt

```diff
@@ -113,7 +115,7 @@ internal class MessageChannel(
             it != "null"
         }
         if (messageId != null && queue.containsKey(messageId)) {
-            queue.remove(messageId)?.send(result)
+            queue.remove(messageId)?.takeIf { !it.isClosedForSend }?.send(result)
```

Comment:
use`trySend`?

Reply:
Maybe not use catch for known types of errors better?

Reply:
`trySend` does not catch, it only return a Result and it may be useful later.
```kotlin
public final override fun trySend(element: E): ChannelResult<Unit> {
        val result = offerInternal(element)
        return when {
            result === OFFER_SUCCESS -> ChannelResult.success(Unit)
            result === OFFER_FAILED -> {
                // We should check for closed token on trySend as well, otherwise trySend won't be linearizable
                // in the face of concurrent close()
                // See https://github.com/Kotlin/kotlinx.coroutines/issues/359
                val closedForSend = closedForSend ?: return ChannelResult.failure()
                ChannelResult.closed(helpCloseAndGetSendException(closedForSend))
            }
            result is Closed<*> -> {
                ChannelResult.closed(helpCloseAndGetSendException(result))
            }
            else -> error("trySend returned $result")
        }
    }
```

Reply:
ok